test/AppNotificationTests/AppNotificationHelpers.h

```diff
@@ -0,0 +1,237 @@
+    using namespace winrt::Windows::Foundation;
+    using namespace winrt::Microsoft::Windows::AppNotifications;
+    using namespace winrt::Windows::Foundation;
+}
+
+namespace AppNotificationHelpers {
+    winrt::AppNotification CreateToastNotification(winrt::hstring message)
+    {
+        winrt::hstring xmlPayload{ L"<toast>" + message + L"</toast>" };
+        return winrt::AppNotification(xmlPayload);
+    }
+
+    winrt::AppNotification CreateToastNotification()
+    {
+        return CreateToastNotification(L"intrepidToast");
+    }
+
+    void VerifyToastStatus(UINT32 expectedToastId, bool expected)
+    {
+        auto retrieveNotificationsAsync{ winrt::AppNotificationManager::Default().GetAllAsync() };
+        auto scope_exit = wil::scope_exit(
+        [&] {
+            retrieveNotificationsAsync.Cancel();
+        });
+
+        VERIFY_ARE_EQUAL(retrieveNotificationsAsync.wait_for(std::chrono::seconds(300)), winrt::AsyncStatus::Completed);
+        scope_exit.release();
+
+        auto notifications{ retrieveNotificationsAsync.get() };
+
+        bool found{ false };
+        for (auto notification : notifications)
+        {
+            if (notification.Id() == expectedToastId)
+            {
+                found = true;
+                break;
+            }
+        }
+        VERIFY_ARE_EQUAL(found, expected);
+    }
+
+    void VerifyToastIsActive(UINT32 expectedToastId)
+    {
+        VerifyToastStatus(expectedToastId, true);
+    }
+
+    void VerifyToastIsInactive(UINT32 expectedToastId)
+    {
+        VerifyToastStatus(expectedToastId, false);
+    }
+
+    winrt::AppNotificationProgressData GetToastProgressData(
+        std::wstring const& status,
+        std::wstring const& title,
+        double const& progressValue,
+        std::wstring const& progressValueString,
+        uint32_t sequenceNumber)
+    {
+        winrt::AppNotificationProgressData progressData{ sequenceNumber };
+        progressData.Status(status);
+        progressData.Title(title);
+        progressData.Value(progressValue);
+        progressData.ValueStringOverride(progressValueString);
+        return progressData;
+    }
+
+    void ProgressResultOperationHelper(winrt::IAsyncOperation<winrt::AppNotificationProgressResult> progressResultOperation, winrt::AppNotificationProgressResult progressResult)
+    {
+        auto scope_exit = wil::scope_exit(
+        [&] {
+            progressResultOperation.Cancel();
+        });
+
+        VERIFY_ARE_EQUAL(progressResultOperation.wait_for(c_timeout), winrt::AsyncStatus::Completed);
+        scope_exit.release();
+
+        VERIFY_ARE_EQUAL(progressResultOperation.GetResults(), progressResult);
+    }
+
+    void VerifyProgressData(const winrt::AppNotificationProgressData& expected, const winrt::AppNotificationProgressData& actual)
+    {
+        if (!expected && !actual)
+        {
+            return;
+        }
+
+        VERIFY_IS_FALSE(!expected || !actual);
```

Comment:
You'll find this exceedingly cryptic and difficult to troubleshoot. Failures will show up as VERIFY_IS_FALSE has an expression value of 'false'. Not very useful.

Plus the trip negative is difficult to follow.

How does this differ from
```
VERIFY_IS_TRUE(!!expected);
VERIFY_IS_TRUE(!!actual);
```

(or even better, `VERIFY_IS_TRUE(expected.isValid())` or whatever, if such a named method exists)

P.S. Worst case if you ever do need to use VERIFY_IS_TRUE/FALSE with an expression, add an explicit message so you can tell the values e.g. `VERIFY_IS_FALSE(!expected || !actual, WEX::Common::String().Format(L"Expected:%s Actual:%s", !!expected ? L"true" : L"false", !!actual ? L"true" : L"false"));`